integration/ballot.go

```diff
@@ -0,0 +1,465 @@
+			K: kbuff,
+			C: cbuff,
+		}
+	}
+
+	return ballot, nil
+}
+
+// cast vote for the load test
+// return a function to provide a common implementation for
+// the load test and the scenario test
+func castVotesLoad(numVotesPerSec, numSec int) func(BallotSize, chunksPerBallot int, formID, contentType string, proxyArray []string, pubKey kyber.Point, secret kyber.Scalar, t *testing.T) []types.Ballot {
+	return (func(BallotSize, chunksPerBallot int, formID, contentType string, proxyArray []string, pubKey kyber.Point, secret kyber.Scalar, t *testing.T) []types.Ballot {
+
+		t.Log("cast ballots")
+
+		//make List of identical valid ballots
+		b1 := string("select:" + encodeIDBallot("bb") + ":0,0,1,0\n" + "text:" + encodeIDBallot("ee") + ":eWVz\n\n") //encoding of "yes"
+
+		numVotes := numVotesPerSec * numSec
+
+		// create all the ballots
+		ballotList := make([]string, numVotes)
+		for i := 1; i <= numVotes; i++ {
+			ballotList[i-1] = b1
+		}
+
+		votesfrontend := make([]types.Ballot, numVotes)
+
+		fakeConfiguration := fake.BasicConfiguration
+
+		for i := 0; i < numVotes; i++ {
+
+			var bMarshal types.Ballot
+			form := types.Form{
+				Configuration: fakeConfiguration,
+				FormID:        formID,
+				BallotSize:    BallotSize,
+			}
+
+			err := bMarshal.Unmarshal(ballotList[i], form)
+			require.NoError(t, err)
+
+			votesfrontend[i] = bMarshal
+		}
+		proxyCount := len(proxyArray)
+
+		// all ballots are identical
+		ballot, err := marshallBallotManual(b1, pubKey, chunksPerBallot)
+		require.NoError(t, err)
+
+		// atomic counter
+		var includedVoteCount uint64
+
+		for i := 0; i < numSec; i++ {
+			// send the votes asynchrounously and wait for the response
+
+			for j := 0; j < numVotesPerSec; j++ {
+				randomproxy := proxyArray[rand.Intn(proxyCount)]
+				castVoteRequest := ptypes.CastVoteRequest{
+					UserID: "user" + strconv.Itoa(i*numVotesPerSec+j),
+					Ballot: ballot,
+				}
+				// cast asynchrounously and increment includedVoteCount
+				// if the cast was succesfull
+				go cast(false, i*numVotesPerSec+j, castVoteRequest, contentType, randomproxy, formID, secret, &includedVoteCount, t)
+
+			}
+			t.Logf("casted votes %d", (i+1)*numVotesPerSec)
+			time.Sleep(time.Second)
+
+		}
+
+		// wait until includedVoteCount == numVotes
+		// i.e. that every votes has been included
+		for {
+			if atomic.LoadUint64(&includedVoteCount) == uint64(numVotes) {
+				break
+			}
+
+			t.Logf("Waiting... included votes %d", atomic.LoadUint64(&includedVoteCount))
+			infos := getFormInfo(proxyArray[0], formID, t)
+			// check that our counter is synchronized with the blockchain
+			t.Logf("Voters count: %v", len(infos.Voters))
+			// check every 10 seconds
+			time.Sleep(time.Second * 10)
+		}
+
+		time.Sleep(time.Second * 30)
+
+		return votesfrontend
+	})
+}
+
+func cast(isRetry bool, idx int, castVoteRequest ptypes.CastVoteRequest, contentType, randomproxy, formID string, secret kyber.Scalar, includedVoteCount *uint64, t *testing.T) {
```

Comment:
Be careful in the order of the parameters so that it makes sense when you call function. For instance, I think the "isRetry" parameter should be just before 't' in the paramater list because it sounds like a question you answer once you defined your call to the function.